code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -1097,5 +1100,13 @@ Pass a positive integer as an argument to override a bot's default speed.
 		I.icon_state = null
 	path.Cut(1, 2)
 
+/mob/living/simple_animal/bot/proc/clear_path_image()
+	if(!path || !length(path))
+		return
+	for(var/P in path)
+		var/image/I = path[P]
+		if(I?.icon_state)
+			I.icon_state = null
```

Comment:
use `as anything` please

and, why cant we delete the paths instead of setting their icon state to null?

Reply:
I tried deleting them but it didn't work. I haven't spent the time to properly understand the underlaying tech to know why not.